### PR TITLE
Model mommy and various

### LINKED ISSUE #590: Editing email of user to existing email causes Internal Server Error
When editing a user and setting the email address to one that is already used by another user in the system causes the following error:

```
Internal Server Error: /staff/user/create Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
psycopg2.IntegrityError: duplicate key value violates unique constraint "evaluation_userprofile_email_key"
DETAIL:  Key (email)=(a@b.com) already exists.
```

The form should check the address during cleaning and show an error in the UI.

### PR DESCRIPTION
**Pull request by karyon**
introduces model_mommy and replaces some fixtures with it, fixes #590, some minor cleanups/fixes.

### CODE REVIEW

**Comment by janno42** [comment]
sorry for the migration conflict ;)

File: evap/staff/forms.py [hunk 1]
```diff
@@ -368,32 +367,41 @@ def __init__(self, *args, **kwargs):
...
+        return self.cleaned_data['first_name'].strip()
+
+    def clean_last_name(self):
+        return self.cleaned_data['last_name'].strip()
 
-        super()._post_clean(*args, **kw)
+    def clean_title(self):
+        return self.cleaned_data['title'].strip()
+
+    def save(self):
```
Comment (janno42) [inline evap/staff/forms.py]: why not ```save(self, *args, **kw)```?

Comment (karyon) [inline evap/staff/forms.py]: because that would be the right thing to do!

**Comment by karyon** [comment]
updated and two more commits added, please review.

File: evap/settings.py [hunk 2]
```diff
@@ -252,6 +252,9 @@ def show_toolbar(request):
         'SHOW_TOOLBAR_CALLBACK': 'evap.settings.show_toolbar',
     }
 
+MOMMY_CUSTOM_FIELDS_GEN = {
```
Comment (janno42) [inline evap/settings.py]: maybe a comment here?

**Comment by janno42** [comment]
looks good now! you could add the one comment in the settings and then... :shipit:

**Pull request merged**